builder/llpcPipeline.h

```diff
@@ -49,6 +49,136 @@ using namespace llvm;
 
 class BuilderContext;
 
+// =====================================================================================================================
+// Structs for setting shader modes, e.g. Builder::SetCommonShaderMode
+
+// FP rounding mode. These happen to have values one more than the corresponding register field in current
+// hardware, so we can make the zero initializer equivalent to DontCare.
+enum class FpRound : uint32_t
+{
+    DontCare,   // Don't care
+    Even,       // Round to nearest even
+    Positive,   // Round up towards positive infinity
+    Negative,   // Round down tiwards negative infinity
+    Zero        // Round towards zero
+};
+
+// Denormal flush mode. These happen to have values one more than the corresponding register field in current
+// hardware, so we can make the zero initializer equivalent to DontCare.
+enum class FpDenorm :uint32_t
```

Comment:
That clashes with an existing enum in llpcIntrinsDefs.h, so I also renamed the llpcIntrinsDefs one to HwFpDenormMode.

Reply:
I think you can rename FpRoundMode to FloatRoundMode and FpDenormMode to FloatDenormMode in llpcIntrinsDefs.h because they both correspond to FLOAT_MODE register field. Don't qualify them with "Hw" because all data definitions in this file are mostly "hardware".

Reply:
Good idea.